fix(replay): fix useReplayList hook

jerryzhou196 opened the pull request:
#79136 fixed a bug, but the hardcoding should be specific to `transactionReplays`, rather than the `fetchReplayList` hook.

jerryzhou196 committed:
fix(replay): fix query property in transactions

jerryzhou196 committed:
Merge branch 'master' into fix-fetch-replay-list-hook

jerryzhou196 committed:
fix(replay): Add enabled option to useReplayList and adjust fetching logic

jerryzhou196 commented on static/app/views/performance/transactionSummary/transactionReplays/useReplaysFromTransaction.tsx:
```diff
@@ -83,16 +83,18 @@ function useReplaysFromTransaction({
     }
 
-    return EventView.fromSavedQuery({
+    const savedQuery: NewQuery = {
```
this is redundant

jerryzhou196 committed:
fix(replay): streamline saved query creation in useReplaysFromTransaction hook

jerryzhou196 reviewed (commented):
self-review

jerryzhou196 committed:
Merge branch 'master' into fix-fetch-replay-list-hook

jerryzhou196 committed:
fix(replay): set default value for enabled option in useReplayList hook

jerryzhou196 committed:
refactor(transactionReplays): simplify empty state test by removing redundant API call and using findByText for assertions

jerryzhou196 committed:
Merge branch 'master' into fix-fetch-replay-list-hook

jerryzhou196 committed:
Merge branch 'master' into fix-fetch-replay-list-hook

ryan953 commented on static/app/utils/replays/hooks/useReplayList.tsx:
```diff
@@ -44,8 +46,11 @@ function useReplayList({
     setData(prev => ({
       ...prev,
-      isFetching: true,
+      isFetching: enabled ? true : false,
```
```suggestion
      isFetching: Boolean(enabled),
```

Nit: This could be written as `Boolean(enabled)` to convert the `undefined` into `false`, instead of using the ternary. 

Using the ternary is a slight extra bit of reading/thinking to make sure that the order of true/false wasn't swapped. If swapping were intended then `!enabled` would be enough.

ryan953 commented on static/app/utils/replays/hooks/useReplayList.tsx:
```diff
@@ -15,6 +15,7 @@ type Options = {
   eventView: EventView;
   location: Location<ReplayListLocationQuery>;
   organization: Organization;
+  enabled?: boolean;
```
Given those other nits about bool, another great approach is to not make optional props whenever possible: Then there won't be undefined in the system!

You will have to go around to all the callsites, which in this case there are very few. Some people might say "oh, I'm just concerned with one callsite or situation, therefore optional prop is easier". But really removing optional stuff keeps the calls understandable over time.

```suggestion
  enabled: boolean;
```

jerryzhou196 replied:
🫡

jerryzhou196 committed:
Merge branch 'master' into fix-fetch-replay-list-hook

jerryzhou196 committed:
fix(replays): Ensure replay list fetching is correctly enabled

Updated the useReplayList hook to enforce fetching behavior based on the enabled flag. The flag is now required in the hook's options, and the fetching state is managed more consistently across components using the hook.

jerryzhou196 committed:
fix(replays): Simplify enabled flag condition for fetching replays

Updated the condition for enabling the replay list fetching in the useReplayList hook to directly check if the eventView.query is not an empty string. This change enhances clarity and maintains consistent fetching behavior.

jerryzhou196 committed:
Merge branch 'master' into fix-fetch-replay-list-hook
